src/main/java/seedu/mealcompanion/command/misc/MakeCommand.java

```diff
@@ -0,0 +1,147 @@
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Removes a specified quantity of an ingredient from the ingredient list
+     *
+     * @param mealCompanionSession the MealCompanionSession containing the list of ingredients
+     * @param quantity the quantity of ingredient to be removed
+     * @param name the name of the ingredient to be removed
+     */
+
+    private void removeIngredient(MealCompanionSession mealCompanionSession, Double quantity, String name) {
+        int indexOfExistingIngredient = findIngredient(mealCompanionSession, name);
+        double fridgeQuantity = mealCompanionSession.getIngredients().get(indexOfExistingIngredient).getQuantity();
+        assert fridgeQuantity >= quantity: "fridgeQuantity should be more than quantity to be removed";
+        double newQuantity = fridgeQuantity - quantity;
+        mealCompanionSession.getIngredients().get(indexOfExistingIngredient).setQuantity(newQuantity);
+        mealCompanionSession.getIngredientStorage().writeIngredientsToFile(mealCompanionSession.getIngredients());
+    }
+
+    /**
+     * Removes the ingredients needed to make the specified recipe from inventory
+     *
+     * @param mealCompanionSession the MealCompanionSession containing the list of ingredients
+     * @param recipe the recipe that is to be made
+     */
+    private void makeRecipe(MealCompanionSession mealCompanionSession, Recipe recipe) {
+        IngredientList recipeIngredients = recipe.getIngredients();
+        for (int i = 0; i < recipeIngredients.size(); i += 1) {
+            double quantityToRemove = recipeIngredients.get(i).getQuantity();
+            String name = recipeIngredients.get(i).getMetadata().getName();
+            removeIngredient(mealCompanionSession, quantityToRemove, name);
+        }
+        mealCompanionSession.getUi().printMessage("Success! The ingredients needed were removed from inventory");
+    }
+
+    public void execute(MealCompanionSession mealCompanionSession) {
+        try {
+            if (this.name.isBlank()) {
+                throw new MealCompanionException("recipe name cannot be blank");
+            }
+            RecipeList recipes = mealCompanionSession.getRecipes();
+            int index = findRecipeName(recipes);
+            if (index == -1) {
+                throw new MealCompanionException("recipe name not found");
+            }
+            Recipe recipe = recipes.getRecipe(index);
+            IngredientList fridgeIngredients = mealCompanionSession.getIngredients();
+            if (canMakeRecipe(recipe, fridgeIngredients)) {
+                makeRecipe(mealCompanionSession, recipe);
+            } else {
+                throw new MealCompanionException("Ingredients in inventory is insufficient");
+            }
+        } catch (Exception e) {
+            System.out.println(e);
+        }
+    }
```

Comment:
Could you include some tests for this function?

Reply:
Added

Reply:
looks good